akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -522,8 +522,9 @@ final case class Fold[In, Out](zero: Out, f: (Out, In) ⇒ Out) extends GraphSta
           pull(in)
         } catch {
           case NonFatal(ex) ⇒ decider(ex) match {
-            case Supervision.Stop ⇒ failStage(ex)
-            case _ ⇒
+            case Supervision.Stop   ⇒ failStage(ex)
```

Comment:
Pull will only generate a failure if the stage is broken (pulling multiple times without push inbetween etc), and then supervision does not matter much anyway.